spec/index.bs

```diff
@@ -1113,6 +1101,38 @@ Given an {{HTMLScriptElement}} (|script|), this algorithm performs the following
 
     If the algorithm threw an error, rethrow the error.
 
+## Get Trusted Types-compliant attribute value ## {#validate-attribute-mutation}
+To <dfn abstract-op export>get Trusted Types-compliant attribute value</dfn> on {{Attr}} |attribute| with {{Element}} |element| and {{TrustedType}} or a string |newValue|, perform the following steps:
+
+1. Let |expectedType| be null.
+1. If |attribute|'s <a for="Attr">namespace</a> is null or is a <a>HTML namespace</a>, find the row in the following table, where |element| is in the first column and |attribute|'s <a for="Attr">local name</a> is an <a>ASCII case-insensitive</a> match for any the string in the
```

Comment:
The check for the HTML namespace here is wrong.

The case-insensitive match seems wrong. Especially if as you say this is IDL-backed in some implementations. Can you point to a test case that exercises that code?

Reply:
> The check for the HTML namespace here is wrong.

Should I just check for null here? I have trouble reverse engineering what a null namespace value means vs HTML namespace in DOM. 
 
> The case-insensitive match seems wrong. Especially if as you say this is IDL-backed in some implementations. Can you point to a test case that exercises that code?

Do you mean wrong or unnecessary? Essentially we need to be able to check when `srCdoC` is the attribute name. Do you mean that can never happen in HTML ns?

https://wpt.fyi/results/trusted-types/Element-setAttribute.html?label=experimental&label=master&aligned (@mbrodesser-Igalia we need to add a mixed-case attribute name test case there).

Essentially I want to do the check for specific html+attr combinations. Assuming I want to future-proof it to add e.g. svg:script+xlink:href, what's the easiest language to get there?

Reply:
An attribute in the HTML namespace would not be recognized as having any meaning. You can test using `setAttributeNS()`. So yeah, just check for null.

Similarly, if using `setAttributeNS()` you set an attribute with no namespace and a local name of `SRCDOC`, it also would not work. Perhaps you still want to block it because of re-parsing concerns? Note that you cannot test this using `setAttribute()`. That lowercases for HTML elements.

Reply:
Got it, thanks. Done. I found that we actually already have a mixed-case attr test in https://github.com/web-platform-tests/wpt/blob/501b929d04/trusted-types/block-string-assignment-to-Element-setAttribute.html#L63, but filed https://github.com/w3c/trusted-types/issues/425 for other cases.

Reply:
As I pointed out that's not a test, right? setAttribute lowercases for HTML elements so it's kind of a useless entry point for testing this.

Reply:
Agreed, #425 proposes a `setAttributeNS` as an entry point.

Reply:
> An attribute in the HTML namespace would not be recognized as having any meaning. You can test using `setAttributeNS()`. So yeah, just check for null.

Indeed.

That's very confusing though, see https://jsfiddle.net/zao4yhc8/3/.

CC @annevk